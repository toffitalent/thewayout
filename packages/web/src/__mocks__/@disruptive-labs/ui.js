const actual = jest.requireActual('@disruptive-labs/ui');

const showToast = jest.fn();

actual.Toast.show = showToast;
actual.Toast.danger = showToast;
actual.Toast.info = showToast;
actual.Toast.success = showToast;
actual.Toast.warning = showToast;

module.exports = {
  ...actual,
  showToast,
};
