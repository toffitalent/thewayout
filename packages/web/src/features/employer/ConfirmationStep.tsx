import { Box, Button, Flex, Text, useWizardContext } from '@disruptive-labs/ui';
import PencilIcon from '@disruptive-labs/ui/dist/icons/Pencil';
import { ReactNode } from 'react';
import { EmployerCreateRequest as WizardData } from '@two/shared';
import styles from './CreatorProfile.module.scss';
import { Steps, stepsConfig } from './profileData';

interface ConfirmationItemProps {
  title: string;
  content: string | ReactNode;
  onClick: () => void;
}

function ConfirmationItem({ title, content, onClick }: ConfirmationItemProps) {
  return (
    <Flex
      direction="row"
      justify="space-between"
      borderBottom={1}
      pv={4}
      className={styles.confirmItem}
    >
      <Flex textAlign="left">
        <Text fontSize="sm" color="grey.500" pb={1}>
          {title}
        </Text>
        <Box fontWeight="700">{content}</Box>
      </Flex>
      <Flex justify="flex-start">
        <Button color="primary" variant="text" onClick={onClick} accessoryLeft={<PencilIcon />}>
          Edit
        </Button>
      </Flex>
    </Flex>
  );
}

export function ConfirmationStep({ onSubmit }: { onSubmit: (data: WizardData) => void }) {
  const { goTo, data } = useWizardContext<WizardData>();

  return (
    <Flex maxWidth={96} ml="auto" mr="auto">
      <Box>
        <ConfirmationItem
          title={stepsConfig[Steps.name].title}
          content={<Text>{data[Steps.name]}</Text>}
          onClick={() => goTo(stepsConfig[Steps.name].id)}
        />
        <ConfirmationItem
          title={stepsConfig[Steps.industry].title}
          content={<Text>{data[Steps.industry]}</Text>}
          onClick={() => goTo(stepsConfig[Steps.industry].id)}
        />
        <ConfirmationItem
          title={stepsConfig[Steps.description].title}
          content={<Text>{data[Steps.description]}</Text>}
          onClick={() => goTo(stepsConfig[Steps.description].id)}
        />
        <ConfirmationItem
          title="Years in Business"
          content={<Text>{data.yearsInBusiness}</Text>}
          onClick={() => goTo(stepsConfig[Steps.details].id)}
        />
        <ConfirmationItem
          title="Number of Employees"
          content={<Text>{data.numberOfEmployees}</Text>}
          onClick={() => goTo(stepsConfig[Steps.details].id)}
        />
        <ConfirmationItem
          title={stepsConfig[Steps.location].title}
          content={
            <>
              <Text>{data.address}</Text>
              <Text>
                {data.city}, {data.state} {data.postalCode}
              </Text>
            </>
          }
          onClick={() => goTo(stepsConfig[Steps.location].id)}
        />
      </Box>
      <Button colorScheme="primary" mt={12} fluid onClick={() => onSubmit(data as WizardData)}>
        Confirm
      </Button>
    </Flex>
  );
}
