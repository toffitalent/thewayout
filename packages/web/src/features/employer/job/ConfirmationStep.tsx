import { Box, Button, Flex, Text, useWizardContext } from '@disruptive-labs/ui';
import CalendarEventIcon from '@disruptive-labs/ui/dist/icons/CalendarEvent';
import CheckIcon from '@disruptive-labs/ui/dist/icons/Check';
import ClockIcon from '@disruptive-labs/ui/dist/icons/Clock';
import CurrencyDollarIcon from '@disruptive-labs/ui/dist/icons/CurrencyDollar';
import InfoCircleIcon from '@disruptive-labs/ui/dist/icons/InfoCircle';
import MapPinIcon from '@disruptive-labs/ui/dist/icons/MapPin';
import PencilIcon from '@disruptive-labs/ui/dist/icons/Pencil';
import StarIcon from '@disruptive-labs/ui/dist/icons/Star';
import UsersIcon from '@disruptive-labs/ui/dist/icons/Users';
import { ReactNode } from 'react';
import { TypeOfWork } from '@two/shared';
import { DetailBadge } from '@app/components/DetailBadge';
import { ExperienceLevel } from '@app/components/ExperienceLevel';
import { job } from '@app/data/jobText';
import { months } from '@app/data/months';
import { formatSalaryRange } from '@app/utils';
import { Steps, stepsConfig } from './createJobData';
import { WizardData } from './createJobDataTypes';
import styles from './Wizard.module.scss';

interface ConfirmationItemProps {
  title?: string | ReactNode;
  content: ReactNode;
  onClick: () => void;
}

function ConfirmationItem({ title, content, onClick }: ConfirmationItemProps) {
  return (
    <Flex justify="space-between" pv={5} className={styles.confirmItem}>
      <Flex textAlign="left">
        {title && (
          <Box fontSize="xl" fontWeight="700" pb={4}>
            {title}
          </Box>
        )}
        {content}
      </Flex>
      <Flex justify="flex-end" className={styles.confirmItemEdit}>
        <Button color="primary" variant="text" onClick={onClick} accessoryLeft={<PencilIcon />}>
          Edit
        </Button>
      </Flex>
    </Flex>
  );
}

export function ConfirmationStep({ onSubmit }: { onSubmit: (data: WizardData) => void }) {
  const { goTo, data, setData } = useWizardContext<WizardData>();

  const details = [
    ...(data.city ? [{ icon: <MapPinIcon />, label: data.city }] : []),
    { icon: <UsersIcon />, label: data.department },
    { icon: <CalendarEventIcon />, label: `From ${months[data.startAtMonth]} ${data.startAtYear}` },
    {
      icon: <CurrencyDollarIcon />,
      label: formatSalaryRange(data.min, data.max, data.salary),
    },
    { icon: <InfoCircleIcon />, label: job[data.typeOfWork] },
    { icon: <ClockIcon />, label: job[data.workingTime] },
    { icon: <StarIcon />, label: `Experience: ${data.experience.toUpperCase()}` },
  ];

  if (data.typeOfWork !== TypeOfWork.onsite && data.address) {
    setData({
      ...data,
      address: undefined,
      city: undefined,
      state: undefined,
      postalCode: undefined,
    });
  }

  return (
    <>
      <Button
        maxWidth={96}
        ml="auto"
        mr="auto"
        colorScheme="primary"
        mt={2}
        fluid
        onClick={() => onSubmit(data)}
      >
        Save
      </Button>
      <Flex border rounded="xl" mt={10} className={styles.confirmWrapper}>
        <Box>
          <ConfirmationItem
            title={data.title}
            content={<Text fontWeight="700">{data.description}</Text>}
            onClick={() => goTo(stepsConfig[Steps.details].id)}
          />
          <ConfirmationItem
            content={
              <Flex direction="row" wrap="wrap" container spacing={4}>
                {details.map((el) => (
                  <Flex item key={el.label}>
                    <DetailBadge {...el} />
                  </Flex>
                ))}
              </Flex>
            }
            onClick={() => goTo(stepsConfig[Steps.details].id)}
          />
          <ConfirmationItem
            title={stepsConfig[Steps.responsibilities].title}
            content={
              <Flex container direction="column" spacing={4}>
                {data.responsibilities.map((el) => (
                  <Flex direction="row" item key={el}>
                    <CheckIcon color="primary" height={6} width={6} mr={4} />
                    <Text>{el}</Text>
                  </Flex>
                ))}
              </Flex>
            }
            onClick={() => goTo(stepsConfig[Steps.responsibilities].id)}
          />
          <ConfirmationItem
            title={
              <Box className={styles.skillsTitle}>
                <Text mr={4} className={styles.skillsTitleText}>
                  {stepsConfig[Steps.skills].title}
                </Text>
                <ExperienceLevel level={data.experience} />
              </Box>
            }
            content={<Text>{data.skillsDescription}</Text>}
            onClick={() => goTo(stepsConfig[Steps.skills].id)}
          />
          <ConfirmationItem
            title={stepsConfig[Steps.pay].title}
            content={
              <>
                <Flex direction="row">
                  <CurrencyDollarIcon color="primary" height={6} width={6} mr={4} />
                  <Text fontWeight="700">{data.salaryDescription}</Text>
                </Flex>
                <Flex container direction="column" spacing={4} mt={1}>
                  {(data.bonuses || []).map((el) => (
                    <Flex direction="row" item key={el}>
                      <CheckIcon color="primary" height={6} width={6} mr={4} />
                      <Text>{el}</Text>
                    </Flex>
                  ))}
                </Flex>
              </>
            }
            onClick={() => goTo(stepsConfig[Steps.pay].id)}
          />
        </Box>
      </Flex>
    </>
  );
}
