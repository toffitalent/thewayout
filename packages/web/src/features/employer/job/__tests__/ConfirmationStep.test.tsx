import { useWizardContext } from '@disruptive-labs/ui';
import { fixtures, render } from '@test';
import { ConfirmationStep } from '../ConfirmationStep';

jest.mock('@disruptive-labs/ui', () => ({
  ...jest.requireActual('@disruptive-labs/ui'),
  useWizardContext: jest.fn(),
}));

describe('ConfirmationStep', () => {
  beforeEach(() => {
    (useWizardContext as any).mockImplementation(() => ({
      data: {
        ...fixtures.job,
        startAtMonth: '2',
        startAtYear: '2022',
        ...fixtures.job.salaryOptions,
      },
      setData: jest.fn(),
      goTo: jest.fn(),
    }));
  });

  test('renders correctly', async () => {
    const { container } = render(<ConfirmationStep onSubmit={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });
});
