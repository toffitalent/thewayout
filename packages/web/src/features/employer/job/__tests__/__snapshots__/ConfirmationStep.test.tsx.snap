// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationStep renders correctly 1`] = `
<div>
  <button
    class="button md solid c-primary fluid maxWidth-96 marginTop-2 marginRight-auto marginLeft-auto"
    type="button"
  >
    Save
  </button>
  <div
    class="confirmWrapper borderTop borderRight borderBottom borderLeft rounded-xl marginTop-10"
  >
    <div
      class=""
    >
      <div
        class="confirmItem justifyContent-space-between paddingTop-5 paddingBottom-5"
      >
        <div
          class="textAlign-left"
        >
          <div
            class="paddingBottom-4 fontSize-xl fontWeight-700"
          >
            Car Wash Attendant
          </div>
          <p
            class="text fontWeight-700"
          >
            Starting wage is $13 per hour + tips, with an Employer Bonus after positive a 90 Day Review (provided by the employer)
          </p>
        </div>
        <div
          class="confirmItemEdit justifyContent-flex-end"
        >
          <button
            class="button md text c-primary color-primary"
            type="button"
          >
            <span
              aria-hidden="true"
              class="accessoryLeft"
            >
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                stroke-width="2"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  stroke="none"
                />
                <path
                  d="M4 20h4L18.5 9.5a2.828 2.828 0 10-4-4L4 16v4M13.5 6.5l4 4"
                />
              </svg>
            </span>
            Edit
          </button>
        </div>
      </div>
      <div
        class="confirmItem justifyContent-space-between paddingTop-5 paddingBottom-5"
      >
        <div
          class="textAlign-left"
        >
          <div
            class="container spacing-4 flexDirection-row flexWrap-wrap"
          >
            <div
              class="item"
            >
              <div
                class=""
              >
                <div
                  class="detailBadge color-primary alignItems-center flexDirection-row display-flex paddingRight-4 paddingLeft-4"
                >
                  <svg
                    aria-hidden="true"
                    class="icon"
                    focusable="false"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      stroke="none"
                    />
                    <path
                      d="M5 7a4 4 0 108 0 4 4 0 10-8 0M3 21v-2a4 4 0 014-4h4a4 4 0 014 4v2M16 3.13a4 4 0 010 7.75M21 21v-2a4 4 0 00-3-3.85"
                    />
                  </svg>
                  <p
                    class="text marginLeft-2"
                  >
                    Customer Service
                  </p>
                </div>
              </div>
            </div>
            <div
              class="item"
            >
              <div
                class=""
              >
                <div
                  class="detailBadge color-primary alignItems-center flexDirection-row display-flex paddingRight-4 paddingLeft-4"
                >
                  <svg
                    aria-hidden="true"
                    class="icon"
                    focusable="false"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      stroke="none"
                    />
                    <path
                      d="M4 7a2 2 0 012-2h12a2 2 0 012 2v12a2 2 0 01-2 2H6a2 2 0 01-2-2zM16 3v4M8 3v4M4 11h16"
                    />
                    <path
                      d="M8 15h2v2H8z"
                    />
                  </svg>
                  <p
                    class="text marginLeft-2"
                  >
                    From Feb. 2022
                  </p>
                </div>
              </div>
            </div>
            <div
              class="item"
            >
              <div
                class=""
              >
                <div
                  class="detailBadge color-primary alignItems-center flexDirection-row display-flex paddingRight-4 paddingLeft-4"
                >
                  <svg
                    aria-hidden="true"
                    class="icon"
                    focusable="false"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      stroke="none"
                    />
                    <path
                      d="M16.7 8A3 3 0 0014 6h-4a3 3 0 000 6h4a3 3 0 010 6h-4a3 3 0 01-2.7-2M12 3v3m0 12v3"
                    />
                  </svg>
                  <p
                    class="text marginLeft-2"
                  >
                    $15 - $30/hour
                  </p>
                </div>
              </div>
            </div>
            <div
              class="item"
            >
              <div
                class=""
              >
                <div
                  class="detailBadge color-primary alignItems-center flexDirection-row display-flex paddingRight-4 paddingLeft-4"
                >
                  <svg
                    aria-hidden="true"
                    class="icon"
                    focusable="false"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      stroke="none"
                    />
                    <path
                      d="M3 12a9 9 0 1018 0 9 9 0 00-18 0M12 9h.01"
                    />
                    <path
                      d="M11 12h1v4h1"
                    />
                  </svg>
                  <p
                    class="text marginLeft-2"
                  >
                    Hybrid
                  </p>
                </div>
              </div>
            </div>
            <div
              class="item"
            >
              <div
                class=""
              >
                <div
                  class="detailBadge color-primary alignItems-center flexDirection-row display-flex paddingRight-4 paddingLeft-4"
                >
                  <svg
                    aria-hidden="true"
                    class="icon"
                    focusable="false"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      stroke="none"
                    />
                    <path
                      d="M3 12a9 9 0 1018 0 9 9 0 00-18 0"
                    />
                    <path
                      d="M12 7v5l3 3"
                    />
                  </svg>
                  <p
                    class="text marginLeft-2"
                  >
                    Full-time
                  </p>
                </div>
              </div>
            </div>
            <div
              class="item"
            >
              <div
                class=""
              >
                <div
                  class="detailBadge color-primary alignItems-center flexDirection-row display-flex paddingRight-4 paddingLeft-4"
                >
                  <svg
                    aria-hidden="true"
                    class="icon"
                    focusable="false"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      stroke="none"
                    />
                    <path
                      d="M12 17.75l-6.172 3.245 1.179-6.873-5-4.867 6.9-1 3.086-6.253 3.086 6.253 6.9 1-5 4.867 1.179 6.873z"
                    />
                  </svg>
                  <p
                    class="text marginLeft-2"
                  >
                    Experience: MID
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="confirmItemEdit justifyContent-flex-end"
        >
          <button
            class="button md text c-primary color-primary"
            type="button"
          >
            <span
              aria-hidden="true"
              class="accessoryLeft"
            >
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                stroke-width="2"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  stroke="none"
                />
                <path
                  d="M4 20h4L18.5 9.5a2.828 2.828 0 10-4-4L4 16v4M13.5 6.5l4 4"
                />
              </svg>
            </span>
            Edit
          </button>
        </div>
      </div>
      <div
        class="confirmItem justifyContent-space-between paddingTop-5 paddingBottom-5"
      >
        <div
          class="textAlign-left"
        >
          <div
            class="paddingBottom-4 fontSize-xl fontWeight-700"
          >
            Responsibilities & Expectations
          </div>
          <div
            class="container spacing-4 flexDirection-column"
          />
        </div>
        <div
          class="confirmItemEdit justifyContent-flex-end"
        >
          <button
            class="button md text c-primary color-primary"
            type="button"
          >
            <span
              aria-hidden="true"
              class="accessoryLeft"
            >
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                stroke-width="2"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  stroke="none"
                />
                <path
                  d="M4 20h4L18.5 9.5a2.828 2.828 0 10-4-4L4 16v4M13.5 6.5l4 4"
                />
              </svg>
            </span>
            Edit
          </button>
        </div>
      </div>
      <div
        class="confirmItem justifyContent-space-between paddingTop-5 paddingBottom-5"
      >
        <div
          class="textAlign-left"
        >
          <div
            class="paddingBottom-4 fontSize-xl fontWeight-700"
          >
            <div
              class="skillsTitle"
            >
              <p
                class="text skillsTitleText marginRight-4"
              >
                Skills & Experience
              </p>
              <div
                class="alignItems-center flexDirection-row display-flex"
              >
                <p
                  class="text marginRight-2 fontSize-xs fontWeight-700"
                >
                  MID
                </p>
                <div
                  class="rounded-3xl bgcolor-primary width-4 height-4 marginRight-1"
                />
                <div
                  class="rounded-3xl bgcolor-primary width-4 height-4 marginRight-1"
                />
                <div
                  class="rounded-3xl bgcolor-primary width-4 height-4 marginRight-1"
                />
                <div
                  class="rounded-3xl bgcolor-grey-200 width-4 height-4 marginRight-1"
                />
                <div
                  class="rounded-3xl bgcolor-grey-200 width-4 height-4 marginRight-1"
                />
              </div>
            </div>
          </div>
          <p
            class="text"
          >
            The most successful and happy people have PMA – a Positive Mental Attitude! We are a huge believer in putting attitude above aptitude. We can always teach you how to do the job, but we can't teach a Positive Mental Attitude... You either have it or you don't!
          </p>
        </div>
        <div
          class="confirmItemEdit justifyContent-flex-end"
        >
          <button
            class="button md text c-primary color-primary"
            type="button"
          >
            <span
              aria-hidden="true"
              class="accessoryLeft"
            >
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                stroke-width="2"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  stroke="none"
                />
                <path
                  d="M4 20h4L18.5 9.5a2.828 2.828 0 10-4-4L4 16v4M13.5 6.5l4 4"
                />
              </svg>
            </span>
            Edit
          </button>
        </div>
      </div>
      <div
        class="confirmItem justifyContent-space-between paddingTop-5 paddingBottom-5"
      >
        <div
          class="textAlign-left"
        >
          <div
            class="paddingBottom-4 fontSize-xl fontWeight-700"
          >
            Pay & Benefits
          </div>
          <div
            class="flexDirection-row"
          >
            <svg
              aria-hidden="true"
              class="icon color-primary width-6 height-6 marginRight-4"
              focusable="false"
              stroke-width="2"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                stroke="none"
              />
              <path
                d="M16.7 8A3 3 0 0014 6h-4a3 3 0 000 6h4a3 3 0 010 6h-4a3 3 0 01-2.7-2M12 3v3m0 12v3"
              />
            </svg>
            <p
              class="text fontWeight-700"
            />
          </div>
          <div
            class="container spacing-4 flexDirection-column marginTop-1"
          >
            <div
              class="item flexDirection-row"
            >
              <svg
                aria-hidden="true"
                class="icon color-primary width-6 height-6 marginRight-4"
                focusable="false"
                stroke-width="2"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  stroke="none"
                />
                <path
                  d="M5 12l5 5L20 7"
                />
              </svg>
              <p
                class="text"
              >
                Additional $100 Bonus for Positive 90 Day Review
              </p>
            </div>
          </div>
        </div>
        <div
          class="confirmItemEdit justifyContent-flex-end"
        >
          <button
            class="button md text c-primary color-primary"
            type="button"
          >
            <span
              aria-hidden="true"
              class="accessoryLeft"
            >
              <svg
                aria-hidden="true"
                class="icon"
                focusable="false"
                stroke-width="2"
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  stroke="none"
                />
                <path
                  d="M4 20h4L18.5 9.5a2.828 2.828 0 10-4-4L4 16v4M13.5 6.5l4 4"
                />
              </svg>
            </span>
            Edit
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
